### DISCUSSION TITLE
Is there a way to find out PID for focused window via IPC?

### DISCUSSION THREAD

**Discussion by kimifish** [i3/i3#5970] [Q&A] [answered]
I want to change active pulseaudio sink for an application depending on whether it's process is responsible for currently focused window or not. The most reliable way to determine which sink input is created by application is PID. Therefore, I need PID of the process executing in current window. Is it possible to get it via IPC or some other way?

**Comment by KebabLord** [comment, marked as answer]
you can use xprop + xdotools to get the pid of focused window
```bash
xprop -id `xdotool getwindowfocus` | grep '_NET_WM_PID' | grep -oE '[[:digit:]]*$'
```
